File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/github/helpers/FileClient.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl.github.helpers;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GithubClient;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Constants;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Language;

import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessResultAssertions.assertFailure;
import static com.optivem.atddaccelerator.templategenerator.systemtest.util.ProcessResultAssertions.assertSuccess;

public class FileClient {

    private final GithubClient client;

    public FileClient (GithubClient client) {
        this.client = client;
    }

    public void verifyPathsExist(String systemLanguage, String systemTestLanguage) {
        verifyPathLanguageExists("monolith-%s", systemLanguage);
        verifyPathLanguageExists( getWorkflowPath(Constants.COMMIT_STAGE_MONOLITH_FORMAT), systemLanguage);

        verifyPathLanguageExists("system-test-%s/docker-compose.yml", systemTestLanguage);
        verifyPathLanguageExists( getWorkflowPath(Constants.LOCAL_ACCEPTANCE_STAGE_TEST_FORMAT), systemTestLanguage);
        verifyPathLanguageExists( getWorkflowPath(Constants.ACCEPTANCE_STAGE_TEST_FORMAT), systemTestLanguage);
        verifyPathLanguageExists( getWorkflowPath(Constants.QA_STAGE_TEST_FORMAT), systemTestLanguage);
        verifyPathLanguageExists( getWorkflowPath(Constants.PROD_STAGE_TEST_FORMAT), systemTestLanguage);
    }

    private void verifyPathExists(String path) {
        var result = client.viewPath(path);
        assertSuccess(result, "Path '" + path + "' should exist.");
    }

    private void verifyPathDoesNotExist(String path) {
        var result = client.viewPath(path);
        assertFailure(result, "Path '" + path + "' should not exist.");
    }

    private void verifyPathLanguageExists(String pathFormat, String language) {
        for(String l : Language.ALL) {
            var path = String.format(pathFormat, l);
            if(l.equals(language)) {
                verifyPathExists(path);
            } else {
                verifyPathDoesNotExist(path);
            }
        }
    }

    private String getWorkflowPath(String stageFormat) {
        return ".github/workflows/" + stageFormat + ".yml";
    }

}

File: console/tests/Dsl/GitHub/Helpers/FileClient.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util;
using static Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process.ProcessResultAssertions;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl.GitHub.Helpers
{
    public class FileClient
    {
        private readonly GithubClient _client;

        public FileClient(GithubClient client)
        {
            _client = client;
        }

        public void VerifyPathsExist(string systemLanguage, string systemTestLanguage)
        {
            VerifyPathLanguageExists("monolith-{0}", systemLanguage);
            VerifyPathLanguageExists(GetWorkflowPath(Constants.CommitStageMonolithFormat), systemLanguage);

            VerifyPathLanguageExists("system-test-{0}/docker-compose.yml", systemTestLanguage);
            VerifyPathLanguageExists(GetWorkflowPath(Constants.LocalAcceptanceStageTestFormat), systemTestLanguage);
            VerifyPathLanguageExists(GetWorkflowPath(Constants.AcceptanceStageTestFormat), systemTestLanguage);
            VerifyPathLanguageExists(GetWorkflowPath(Constants.QaStageTestFormat), systemTestLanguage);
            VerifyPathLanguageExists(GetWorkflowPath(Constants.ProdStageTestFormat), systemTestLanguage);
        }

        private void VerifyPathExists(string path)
        {
            var result = _client.ViewPath(path);
            result.ShouldSucceed($"Path '{path}' should exist.");
        }

        private void VerifyPathDoesNotExist(string path)
        {
            var result = _client.ViewPath(path);
            result.ShouldFail($"Path '{path}' should not exist.");
        }

        private void VerifyPathLanguageExists(string pathFormat, string language)
        {
            foreach (var l in LanguageExtensions.GetAll())
            {
                var path = string.Format(pathFormat, l);
                if (l == language)
                {
                    VerifyPathExists(path);
                }
                else
                {
                    VerifyPathDoesNotExist(path);
                }
            }
        }

        private string GetWorkflowPath(string stageFormat)
        {
            return $".github/workflows/{stageFormat}.yml";
        }
    }
}